tests/unit_test/channels/whatsapp_test.py

```
@@ -690,4 +691,578 @@
                     }
                 }
                 assert log["status"] == "captured"
-                assert log["recipient"] == "919515991234"
+                assert log["recipient"] == "919515991234"
+
+    @pytest.mark.asyncio
+    async def test_whatsapp_valid_statuses_with_sent_request(self):
+        from kairon.chat.handlers.channels.whatsapp import Whatsapp, WhatsappBot
+        with patch.object(WhatsappBot, "mark_as_read"):
+            with patch.object(Whatsapp, "process_message") as mock_message:
```

Ruff

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)